utils/src/test/java/io/atomix/utils/concurrent/ExceptionHandlingSingleThreadContextTest.java

```diff
@@ -0,0 +1,35 @@
+package io.atomix.utils.concurrent;
+
+import static io.atomix.utils.concurrent.Threads.namedThreads;
+import static org.junit.Assert.assertEquals;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExceptionHandlingSingleThreadContextTest {
+
+  private final Logger log = LoggerFactory.getLogger("thread");
+  private Consumer<Throwable> exceptionHandler;
+  private final ExceptionHandlingSingleThreadContext threadContext =
+      new ExceptionHandlingSingleThreadContext(
+          namedThreads("test", log), e -> exceptionHandler.accept(e));
+
+  @Test
+  public void shouldInvokeHandlerOnException() {
+    CountDownLatch latch = new CountDownLatch(1);
+    exceptionHandler = e -> latch.countDown();
+    threadContext.execute(
+        () -> {
+          throw new RuntimeException();
+        });
+    try {
+      latch.await(2, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
```

Comment:
You can add to the test a throws clause.

And please separate the test in // given, when then blocks